execution_engine/src/core/execution/executor.rs

```diff
@@ -143,6 +143,10 @@ impl Executor {
         let gas_counter: Gas = Gas::default();
         let transfers = Vec::default();
 
+        // Snapshot of the journal before execution, so in case of error
+        // only nonce update can be returned.
```

Comment:
We don't have nonces...

Reply:
That comment was originally there.
This PR had removed it and I just brought it back.
Should I remove it?

Reply:
Yes, let's remove it and replace with something like:
```
so in case of an error we don't apply changes of the execution but keep the payment code effects.
```

Reply:
Done.